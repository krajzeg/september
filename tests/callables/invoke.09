# Test for built-ins
print.invoke(arguments: ["This", "should", "print", "correctly."])

# Test for interpreted functions
sumItUp := |a,b| { a+b }
print("Using invoke() with 'arguments', 2 + 3 seems to be:", sumItUp.invoke(arguments: [2,3]))

# Test for this override
knowThyself := { this }
print("If 'target' overriding works, we should see 42:", knowThyself.invoke(target: 42))

# Test for scope override
scope := [[ a: 17, b: 25 ]]
print("If 'scope' overriding works, we should see 42:", sumItUp.invoke(scope: scope))

# Test for invoke on bound methods
holder := [[]]
holder:::setMe = |value| { this::theAnswer = value }
universe := [[]]
holder.setMe.invoke(target: universe, arguments: [42])
print("If everything went well, 42 again:", universe.theAnswer)
