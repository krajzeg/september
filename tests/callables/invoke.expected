This should print correctly.
Using invoke() with 'arguments', 2 + 3 seems to be: 5
If 'target' overriding works, we should see 42: 42
If 'scope' overriding works, we should see 42: 42
If everything went well, 42 again: 42
